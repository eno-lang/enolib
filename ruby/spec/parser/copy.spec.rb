input = <<~DOC.strip
template:
key<template
key<<template
key < template
key << template
key     < template
key     << template
     key      < template
     key      << template
DOC

describe Enolib::Parser do
  describe 'Copy operations' do
    context = Enolib::Context.new(input)

    it 'parses as expected' do
      expect(context.document).to match_snapshot
    end
  end
end
