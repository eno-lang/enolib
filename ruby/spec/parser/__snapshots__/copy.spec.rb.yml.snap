---
"./spec/parser/copy.spec.rb[1:1:1]": &1
  :elements:
  - :index: 0
    :line: 0
    :ranges:
      :line:
      - 0
      - 9
      :element_operator:
      - 8
      - 9
      :key:
      - 0
      - 8
    :key: template
    :type: :field_or_fieldset_or_list
    :parent: *1
  - :index: 10
    :line: 1
    :ranges:
      :line:
      - 10
      - 22
      :copy_operator:
      - 13
      - 14
      :template:
      - 14
      - 22
      :key:
      - 10
      - 13
    :template: template
    :type: :field_or_fieldset_or_list
    :key: key
    :parent: *1
  - :index: 23
    :line: 2
    :ranges:
      :line:
      - 23
      - 36
      :copy_operator:
      - 26
      - 28
      :template:
      - 28
      - 36
      :key:
      - 23
      - 26
    :template: template
    :type: :field_or_fieldset_or_list
    :key: key
    :parent: *1
  - :index: 37
    :line: 3
    :ranges:
      :line:
      - 37
      - 51
      :copy_operator:
      - 41
      - 42
      :template:
      - 43
      - 51
      :key:
      - 37
      - 40
    :template: template
    :type: :field_or_fieldset_or_list
    :key: key
    :parent: *1
  - :index: 52
    :line: 4
    :ranges:
      :line:
      - 52
      - 67
      :copy_operator:
      - 56
      - 58
      :template:
      - 59
      - 67
      :key:
      - 52
      - 55
    :template: template
    :type: :field_or_fieldset_or_list
    :key: key
    :parent: *1
  - :index: 68
    :line: 5
    :ranges:
      :line:
      - 68
      - 86
      :copy_operator:
      - 76
      - 77
      :template:
      - 78
      - 86
      :key:
      - 68
      - 71
    :template: template
    :type: :field_or_fieldset_or_list
    :key: key
    :parent: *1
  - :index: 87
    :line: 6
    :ranges:
      :line:
      - 87
      - 106
      :copy_operator:
      - 95
      - 97
      :template:
      - 98
      - 106
      :key:
      - 87
      - 90
    :template: template
    :type: :field_or_fieldset_or_list
    :key: key
    :parent: *1
  - :index: 107
    :line: 7
    :ranges:
      :line:
      - 107
      - 131
      :copy_operator:
      - 121
      - 122
      :template:
      - 123
      - 131
      :key:
      - 112
      - 115
    :template: template
    :type: :field_or_fieldset_or_list
    :key: key
    :parent: *1
  - :index: 132
    :line: 8
    :ranges:
      :line:
      - 132
      - 157
      :copy_operator:
      - 146
      - 148
      :template:
      - 149
      - 157
      :key:
      - 137
      - 140
    :template: template
    :type: :field_or_fieldset_or_list
    :key: key
    :parent: *1
  :type: :document
